File: typescript/examples/get-player-progression.ts
import { MythicMC } from '../src/index.ts'

const api = new MythicMC({ apiKey: process.env.MYTHICMC_API_KEY!, baseUrl: process.env.MYTHICMC_API_URL })
const player = process.argv[2] ?? 'Vicente_1313'

const progression = await api.getPlayerProgression(player)

console.log(`level ${progression.level}, ${progression.experience} XP in total`)

// Progress and target are XP within the current level.
console.log(progression.maxed ? 'max level' : `${progression.progress} / ${progression.target} XP to the next level`)

const achievements = progression.achievements
console.log('achievements:', achievements ? `${achievements.filter(a => a.complete).length} of ${achievements.length}` : 'unknown')

File: python/examples/get_player_progression.py
import os
import sys

from mythicmc import DEFAULT_BASE_URL, MythicMC

key = os.environ["MYTHICMC_API_KEY"]
base_url = os.environ.get("MYTHICMC_API_URL", DEFAULT_BASE_URL)
player = sys.argv[1] if len(sys.argv) > 1 else "Vicente_1313"

with MythicMC(key, base_url=base_url) as api:
    progression = api.get_player_progression(player)

print(f"level {progression.level}, {progression.experience} XP in total")

# Progress and target are XP within the current level.
print("max level" if progression.maxed else f"{progression.progress} / {progression.target} XP to the next level")

achievements = progression.achievements
print("achievements:", f"{sum(a.complete for a in achievements)} of {len(achievements)}" if achievements else "unknown")
